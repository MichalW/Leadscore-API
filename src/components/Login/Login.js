import React from 'react';
import {
  Button, Container, Form, Grid, Input, Label,
} from 'semantic-ui-react';

const Login = () => (
  <Container>
    <Grid columns={3} centered>
      <Grid.Row>
        <Grid.Column>
          <Form>
            <Form.Field>
              <Label>Login</Label>
              <Input placeholder="Login" />
            </Form.Field>

            <Form.Field>
              <Label>Password</Label>
              <input placeholder="Password" type="password" />
            </Form.Field>

            <Button type="submit">Login</Button>
          </Form>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </Container>
);

export default Login;
