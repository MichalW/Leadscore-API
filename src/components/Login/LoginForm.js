import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Label, Segment,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = (event) => {
    this.setState({username: event.target.value});
  };

  onChangePassword = (event) => {
    this.setState({password: event.target.value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field>
            <Label>Login</Label>
            <Input placeholder="Login" onChange={this.onChangeUsername} value={username} />
          </Form.Field>

          <Form.Field>
            <Label>Password</Label>
            <input placeholder="Password" type="password" onChange={this.onChangePassword} value={password} />
          </Form.Field>

          <Button type="submit" fluid primary>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
