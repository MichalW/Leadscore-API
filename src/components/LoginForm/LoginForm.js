import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Label, Segment, Message,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = ({target: {value}}) => {
    this.setState({username: value});
  };

  onChangePassword = ({target: {value}}) => {
    this.setState({password: value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {error, loading} = this.props;
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field>
            <Label>Login</Label>
            <Input placeholder="Login" onChange={this.onChangeUsername} value={username} />
          </Form.Field>

          <Form.Field>
            <Label>Password</Label>
            <input placeholder="Password" type="password" onChange={this.onChangePassword} value={password} />
          </Form.Field>

          {error && (
            <Message negative>
              <p>Invalid credentials</p>
            </Message>
          )}

          <Button type="submit" fluid primary loading={loading}>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  error: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
