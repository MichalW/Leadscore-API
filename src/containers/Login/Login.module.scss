@import "../../styles/colors";

.wrapper {
  background-color: $color-blue;
}
