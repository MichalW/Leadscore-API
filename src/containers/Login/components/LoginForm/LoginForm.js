import React, {PureComponent} from 'react';
import {
  Button, Form, Input, Message, Segment,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class LoginForm extends PureComponent {
  state = {
    username: '',
    password: '',
  };

  onChangeUsername = ({target: {value}}) => {
    this.setState({username: value});
  };

  onChangePassword = ({target: {value}}) => {
    this.setState({password: value});
  };

  onSubmit = () => {
    const {signInRequest} = this.props;
    const {username, password} = this.state;

    signInRequest(username, password);
  };

  render() {
    const {error, loading} = this.props;
    const {username, password} = this.state;

    return (
      <Form onSubmit={this.onSubmit}>
        <Segment>
          <Form.Field
            control={Input}
            label="Login"
            onChange={this.onChangeUsername}
            placeholder="Login"
            required
            value={username}
          />

          <Form.Field
            control={Input}
            label="Password"
            onChange={this.onChangePassword}
            placeholder="Password"
            required
            type="password"
            value={password}
          />

          {error && (
            <Message negative>
              <p>Invalid credentials</p>
            </Message>
          )}

          <Button type="submit" fluid primary loading={loading}>Login</Button>
        </Segment>
      </Form>
    );
  }
}

LoginForm.propTypes = {
  error: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
  signInRequest: PropTypes.func.isRequired,
};

export default LoginForm;
