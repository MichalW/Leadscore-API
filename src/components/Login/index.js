import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import Login from './Login';
import {signInRequest} from '../../actions/authentication';

const mapDispatchToProps = dispatch => ({
  signInRequest: bindActionCreators(signInRequest, dispatch),
});

export default connect(null, mapDispatchToProps)(Login);
