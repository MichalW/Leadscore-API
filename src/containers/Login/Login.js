import React from 'react';
import {Grid, Header} from 'semantic-ui-react';

import LoginForm from '../../components/LoginForm';
import styles from './Login.module.scss';

const Login = () => (
  <Grid
    centered
    className={styles.wrapper}
    columns={3}
    style={{minHeight: '100vh'}}
    verticalAlign="middle"
  >
    <Grid.Row>
      <Grid.Column>
        <Header as="h1">
          <strong>Lead</strong>
          score
        </Header>

        <LoginForm />
      </Grid.Column>
    </Grid.Row>
  </Grid>
);

export default Login;
