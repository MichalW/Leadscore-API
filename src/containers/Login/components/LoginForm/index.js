import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';

import LoginForm from './LoginForm';
import {signInRequest} from '../../../../actions/authentication';

const mapStateToProps = ({authentication: {error, loading}}) => ({
  error,
  loading,
});

const mapDispatchToProps = dispatch => ({
  signInRequest: bindActionCreators(signInRequest, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm);
