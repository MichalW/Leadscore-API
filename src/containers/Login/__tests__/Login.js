import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import Login from '../Login';
import LoginForm from '../../../components/LoginForm';

it('renders Login without crashing', () => {
  const props = {
    username: 'ExampleUsername',
  };

  const login = shallow(<Login {...props} />);

  expect(login.find(LoginForm).exists()).toBe(true);
  expect(toJson(login)).toMatchSnapshot();
});
