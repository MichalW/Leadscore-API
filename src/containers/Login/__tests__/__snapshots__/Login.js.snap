// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login without crashing 1`] = `
<Grid
  centered={true}
  className="wrapper"
  columns={3}
  style={
    Object {
      "minHeight": "100vh",
    }
  }
  verticalAlign="middle"
>
  <GridRow>
    <GridColumn>
      <Header
        as="h1"
      >
        <strong>
          Lead
        </strong>
        score
      </Header>
      <Connect(LoginForm) />
    </GridColumn>
  </GridRow>
</Grid>
`;
